### PR TITLE
chore: align clients

### LINKED ISSUE #204: `make gen-sync` is generating code in conflict with `make format`.
If I run `make gen-sync` some test files get reformatted. `make format` removes the formatting. This can result in changes to files that the commit never touches.

Perhaps `gen-sync` should run `format`?

```
$ git status
On branch issue/#174
Changes to be committed:
  (use "git restore --staged <file>..." to unstage)
	modified:   mypy.ini
	modified:   src/momento/internal/aio/_scs_data_client.py
	new file:   src/momento/internal/common/_data_client_list_ops.py
	modified:   src/momento/internal/synchronous/_scs_data_client.py
	modified:   src/momento/responses/__init__.py
	new file:   src/momento/responses/list_data/__init__.py
	new file:   src/momento/responses/list_data/list_fetch.py
	modified:   src/momento/simple_cache_client_async.py

$ make gen-sync

$ git status
On branch issue/#174
Changes to be committed:
  (use "git restore --staged <file>..." to unstage)
	modified:   mypy.ini
	modified:   src/momento/internal/aio/_scs_data_client.py
	new file:   src/momento/internal/common/_data_client_list_ops.py
	modified:   src/momento/internal/synchronous/_scs_data_client.py
	modified:   src/momento/responses/__init__.py
	new file:   src/momento/responses/list_data/__init__.py
	new file:   src/momento/responses/list_data/list_fetch.py
	modified:   src/momento/simple_cache_client_async.py

Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git restore <file>..." to discard changes in working directory)
	modified:   tests/momento/simple_cache_client/shared_behaviors.py
	modified:   tests/momento/simple_cache_client/test_control.py
	modified:   tests/momento/simple_cache_client/test_init.py

$ make format
reformatted tests/momento/simple_cache_client/test_init.py
reformatted tests/momento/simple_cache_client/shared_behaviors.py
reformatted tests/momento/simple_cache_client/test_control.py

All done! ✨ 🍰 ✨
3 files reformatted, 72 files left unchanged.
Fixing /Volumes/Momento/client-sdk-python/src/momento/internal/synchronous/_scs_data_client.py
Skipped 3 files

$ git status
On branch issue/#174
Changes to be committed:
  (use "git restore --staged <file>..." to unstage)
	modified:   mypy.ini
	modified:   src/momento/internal/aio/_scs_data_client.py
	new file:   src/momento/internal/common/_data_client_list_ops.py
	modified:   src/momento/internal/synchronous/_scs_data_client.py
	modified:   src/momento/responses/__init__.py
	new file:   src/momento/responses/list_data/__init__.py
	new file:   src/momento/responses/list_data/list_fetch.py
	modified:   src/momento/simple_cache_client_async.py
```

### PR DESCRIPTION
**Pull request by malandis**
We update the async client and codegen module to generate the
synchronous client from the async one.

Updates the Makefile target to always generate the synchronous client from the asynchronous one.

Also refactors the `gen-sync` Makefile target to run format.

Closes #204

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -24,18 +24,20 @@ lint:
...
-#	@poetry run python -m momento.internal.codegen src/momento/simple_cache_client_async.py src/momento/simple_cache_client.py
+	@poetry run python -m momento.internal.codegen src/momento/simple_cache_client_async.py src/momento/simple_cache_client.py
 	@poetry run python -m momento.internal.codegen tests/momento/simple_cache_client/shared_behaviors_async.py tests/momento/simple_cache_client/shared_behaviors.py 
 	@poetry run python -m momento.internal.codegen tests/momento/simple_cache_client/test_init_async.py tests/momento/simple_cache_client/test_init.py
 	@poetry run python -m momento.internal.codegen tests/momento/simple_cache_client/test_control_async.py tests/momento/simple_cache_client/test_control.py
 	@poetry run python -m momento.internal.codegen tests/momento/simple_cache_client/test_scalar_async.py tests/momento/simple_cache_client/test_scalar.py
 
+.PHONY: gen-sync
+## Generate synchronous code and tests from asynchronous code.
+gen-sync: do-gen-sync format
```
Comment (schwern) [inline Makefile:39]: ❤️

File: poetry.lock [hunk 2]
```diff
@@ -119,64 +119,61 @@ pyflakes = ">=2.5.0,<2.6.0"
 
 [[package]]
 name = "grpcio"
-version = "1.50.0"
+version = "1.51.1"
```
Comment (schwern) [inline poetry.lock:122]: Is regenerating the lockfile necessary?

File: src/momento/simple_cache_client.py [hunk 3]
```diff
@@ -48,8 +50,22 @@
...
+`
+    In the javascript SDK, multiple clients resulted in an obvious performance improvement.
+    However, in the python SDK I have not yet been able to observe a clear benefit.
+    So for now, we are putting the plumbing in place so that we can more easily test performance
+    with multiple connections in the future, but we are leaving the default value set to 1.
+
+    We are hard-coding the value for now, because we haven't yet designed the API for
+    users to use to configure tunables:
+    https://github.com/momentohq/dev-eco-issue-tracker/issues/85
+    """
```
Comment (schwern) [inline src/momento/simple_cache_client.py:65]: Should this be user-visible documentation?

File: poetry.lock (hunk 2 above)
Comment (malandis) [reply poetry.lock:122]: My local setup complained it was out of sync

File: src/momento/simple_cache_client.py (hunk 3 above)
Comment (malandis) [reply src/momento/simple_cache_client.py:65]: The field is internal so it's OK.

Comment (schwern) [reply src/momento/simple_cache_client.py:65]: It won't be visible in generated docs?

Comment (schwern) [reply src/momento/simple_cache_client.py:65]: If that's acceptable doc style, ok.

**Review by schwern — APPROVED** [review]

**Pull request merged**